pkg/clustermesh/mcsapi/types/mcsapiservicespec_test.go

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsapiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+func TestMCSAPIServiceSpec(t *testing.T) {
+	exportTime := metav1.Now().Rfc3339Copy()
+	exportTime.Time = exportTime.Local()
```

Comment:
Any reason why this is required? It seems to imply that the test is somehow timezone dependent. Same for the other occurrences.

Reply:
Yep so that's because the marshal function of time from Kubernetes does this: https://github.com/cilium/cilium/blob/main/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L136 and as we do not call the Equal method of the time object to compare but use the `require.Equal` we compare byte by byte and that's why we need it. If we would call the `Equal` method  from the time object we won't need it though

Reply:
Fair enough, could you please just add a comment in that case? I had tried removing it, and hit no issues on my side, hence why I was asking whether it was necessary.

Reply:
Yep will add a comment :+1:. FYI it also works on my laptop but didn't in the CI actually, I am not entirely sure what's different though but it means that adding this would very likely make it works on more stuff